### PR TITLE
Market lady ping pong fix

### PR DESCRIPTION
**Pull request by PrettyFlower**
Save file: [market lady ping pong.zip](https://github.com/Keriew/augustus/files/12502461/market.lady.ping.pong.zip)

Just plop down a market in the middle block and set it to only get pottery. The market lady gets stuck walking back and forth.

pingpongcropped

### CODE REVIEW

File: src/figuretype/supplier.c [hunk 1]
```diff
@@ -168,6 +168,27 @@ static int change_market_supplier_destination(figure *f, int dst_building_id)
     return 1;
 }
 
+static int new_market_dest_better_than_old_market_dest(figure *f, resource_type r, resource_storage_info info[RESOURCE_MAX])
```
Comment (crudelios) [inline src/figuretype/supplier.c:171]:
```suggestion
static int is_better_destination(figure *f, resource_type r, resource_storage_info *info)
```

You only need to return the pointer to the information for the resource `r`, not the whole array.

File: src/figuretype/supplier.c [hunk 2]
```diff
@@ -168,6 +168,27 @@ static int change_market_supplier_destination(figure *f, int dst_building_id)
...
+        return 1;
+    } else if (old_dest->type == BUILDING_GRANARY && old_dest->resources[r] <= 0) {
+        return 1;
+    } else if (old_dest->type == BUILDING_WAREHOUSE && building_warehouse_get_amount(old_dest, r) <= 0) {
+        return 1;
+    }
+    // make sure the new building is less than or equal to half the distance from the old
+    // building to help prevent market ladies from "ping ponging" back and forth
+    int old_dest_dist = building_dist(f->x, f->y, 1, 1, old_dest);
+    int half_old_dest_dist = old_dest_dist > 0 ? old_dest_dist / 2 : 0;
```
Comment (crudelios) [inline src/figuretype/supplier.c:185]:
```suggestion
    int half_old_dest_dist = old_dest_dist / 2;
```

No need for the condition, a division from `0` is safe.

In fact you don't even need the variable, just place `old_dest_dist / 2` on the `if` itself.

**Review by crudelios — COMMENTED** [review]
Thanks for this!

Just some small formatting suggestions.

File: src/figuretype/supplier.c (hunk 2 above)
Comment (PrettyFlower) [reply src/figuretype/supplier.c:185]: Oh wow I'm dumb.

**Pull request merged**